### ISSUE TITLE
Gibberish output andquestion on how to increase used threads

### ISSUE THREAD

**Issue by candymint23** [cocktailpeanut/dalai#86]
Hi there, I'm getting some gibbersh outputs even if I lower the rep penalty or retain it to the default one. Also how can we use more threads of the cpu to more than 4? Thank you.
![qqw](https://user-images.githubusercontent.com/113113106/226088922-fcdf0a22-3eb7-4eee-b0a8-0531af946478.png)

**Comment by occultknowledge** [comment]
Yes, I'm also getting strange characters. Is there a fix>? I'm on Windows

**Comment by lseder** [comment]
check if you have complete file - c:\Users\<username>\dalai\alpaca\models\7B\ggml-model-q4_0.bin - 4.2 Gb

**Comment by candymint23** [comment]
> check if you have complete file - c:\Users<username>\dalai\alpaca\models\7B\ggml-model-q4_0.bin - 4.2 Gb

I have it on Llama, btw, I'm also having trouble with it using Alpaca, how can we switch to using it?

**Comment by yellowbikewithbluehandles** [comment]
> check if you have complete file - c:\Users<username>\dalai\alpaca\models\7B\ggml-model-q4_0.bin - 4.2 Gb

Mine is 3.9 gigs, thats so weird I got a "installation done" but I'm getting non sensical answers and characters. How do I make it download the full 4.2 gigs?

**Comment by lseder** [comment]
i have deleted this file manualy and run again "npx dalai alpaca install 7B".

**Comment by candymint23** [comment]
looks like dalai changed the way models worked, managed to rename the file and transfer the alpaca file to the folder specified and it worked. Thanks, now how can we change the threads?

> check if you have complete file - c:\Users<username>\dalai\alpaca\models\7B\ggml-model-q4_0.bin - 4.2 Gb

**Comment by jay5656** [comment]
> check if you have complete file - c:\Users<username>\dalai\alpaca\models\7B\ggml-model-q4_0.bin - 4.2 Gb

mine is 4.1GB and doing this. does it have to be exactly 4.2?

**Comment by KaruroChori** [comment]
The issue is that the value is hardcoded in the queries of the webui, and not exposed like other parameters. If you used a direct command like:
`./main --seed -1 --threads 28 --n_predict 200 --model models/7B/ggml-model-q4_0.bin --top_k 40 --top_p 0.9 --temp 0.1 --repeat_last_n 64 --repeat_penalty 1.3 -p "The power of friendship is overrated."`
it will use 28 threads for example.

It should be trivial to change the code, but it would be better if the default value automatically select all threads available, and not just a fixed amount.

**Comment by TheFlymo** [comment]
Find and edit index.ejs to set the defaults (on my system it was under /root/.npm/_npx/3c737cbb02d79cc9/node_modules/dalai/bin/web/views/index.ejs )
